internal/controller/drplacementcontrol_controller.go

```diff
@@ -166,13 +178,21 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 		// then the DRPC should be deleted as well. The least we should do here is to clean up DPRC.
 		err := r.processDeletion(ctx, drpc, placementObj, logger)
 		if err != nil {
-			logger.Info(fmt.Sprintf("Error in deleting DRPC: (%v)", err))
-
 			statusErr := r.setDeletionStatusAndUpdate(ctx, drpc)
 			if statusErr != nil {
 				err = fmt.Errorf("drpc deletion failed: %w and status update failed: %w", err, statusErr)
+
+				return ctrl.Result{}, err
 			}
 
+			// Is this an expected condition?
+			if errorswrapper.Is(err, deleteInProgress{}) {
+				logger.Info("Deleting DRPC in progress", "reason", err)
+
+				return ctrl.Result{Requeue: true}, nil
+			}
+
+			// Unexpected error.
 			return ctrl.Result{}, err
```

Comment:
I mean just let this be returned and remove the block starting from line 189 to 195 inclusive.

Reply:
The result will be ERROR log with a stacktrace - we see 18 of these for every delete operation. Since this is an expected condition, it is wrong to treat this as an error.

Reply:
> Since this is an expected condition, it is wrong to treat this as an error.

Same as my comment above